src/gsy_e/models/strategy/energy_parameters/forward_profile.py

```diff
@@ -0,0 +1,192 @@
+            {time: value} items.
+        """
+        data = defaultdict(dict)
+        with open(cls.FILENAME, "r", encoding="utf-8") as inf:
+            reader = csv.DictReader(inf, fieldnames=cls.FIELDNAMES, delimiter=";")
+            next(reader)  # Skip header line
+            for row in reader:
+                # Only keep time, discarding the date information to make the profile more general
+                slot = pendulum.parse(row["INTERVAL"], exact=True)
+                # The SSP is expressed in power (kW) units, so we convert it to energy (kWh)
+                for month_idx, month in cls._enumerate_months():
+                    data[month_idx][slot] = float(row[month]) / cls.SLOTS_IN_ONE_HOUR
+
+        return dict(data)
+
+    @classmethod
+    def _enumerate_months(cls):
+        return enumerate(cls.MONTHS, start=1)
+
+
+class ForwardTradeProfileGenerator:
+    """
+    Class to generate a new profile that spreads the trade energy across multiple market slots.
+
+    The values of this profile are obtained by scaling the values of the Standard Solar Profile.
+    """
+
+    _STANDARD_SOLAR_PROFILE = StandardProfileParser.parse()
+
+    def __init__(self, peak_kWh: float):
+        self._scaler = ProfileScaler(self._STANDARD_SOLAR_PROFILE)
+        self._scaled_capacity_profile = self._scaler.scale_by_peak(peak_kWh=peak_kWh)
+
+    def generate_trade_profile(
+            self, energy_kWh: float, market_slot: pendulum.DateTime,
+            product_type: AvailableMarketTypes):
+        """Generate a profile based on the market slot and the product type."""
+        assert energy_kWh > 0, f"The traded energy must be positive. energy_kWh: {energy_kWh}"
+
+        trade_profile = self._scaler.scale_by_peak(peak_kWh=energy_kWh)
+
+        # This subtraction is done _before_ expanding the slots to improve performance
+        # residual_energy_profile = self._subtract_profiles(
+        #     self._scaled_capacity_profile, trade_profile)
+
+        # Target a specific market slot based on the product type and market_slot.
+        return self._expand_profile_slots(trade_profile, market_slot, product_type)
+
+    @staticmethod
+    def _expand_profile_slots(
+            profile: Dict[int, Dict[pendulum.Time, float]], market_slot: pendulum.DateTime,
+            product_type: AvailableMarketTypes):
+        """
+        Create all the profile slots targeting a specific period of time.
+
+        The period depends on the type of the product (e.g. yearly/monthly/etc.).
+
+        Return:
+            A new profile complete with all time slots. E.g.:
+                {
+                    <time_slot>: value,
+                    ...
+                }
+        """
+        assert product_type in ALLOWED_MARKET_TYPES
+
+        period = None
+        if product_type == AvailableMarketTypes.YEAR_FORWARD:
+            period = pendulum.period(
+                start=market_slot.start_of("year"), end=market_slot.end_of("year"))
```

Comment:
I had a look into the documentation of pendulum and AFAIU, you need to have a datetime object in order for the `start_of` and `end_of` methods to work properly. Think about leap years or the length of the months. You somehow have to tell pendulum which month you want the end of for. Or did I mis a step?

Reply:
I think that market_slot is a DateTime object, that's why it works that way here.

Reply:
Ahh, I oversaw this. My attention was towards `pendulum.period`. This should work. Thanks!